loleaflet/src/core/Socket.js

```diff
@@ -268,6 +268,30 @@ L.Socket = L.Class.extend({
 			     'background:#ddf;color:black', color, 'color:black');
 	},
 
+	// The problem: if we process one websocket message at a time, the
+	// browser -loves- to trigger a re-render as we hit the main-loop,
+	// this takes ~200ms on a large screen, and worse we get
+	// producer/consumer issues that can fill a multi-second long
+	// buffer of web-socket messages in the client that we can't
+	// process so - slurp and the emit at idle - its faster to delay!
+	_slurpMessage: function(e) {
+		var that = this;
+		if (!this._slurpQueue || !this._slurpQueue.length) {
+			setTimeout(function() {
```

Comment:
We should cancel existing timer events and append to _slurpQueue if one still exists. I fear this can result in multiple timer events, which wouldn't be a good thing.

We can store the timer object in 'this' and cancel it if non-null before calling setTimeout and resetting it.

Reply:
we only add a timer if the slurp queue is empty - so that seems unlikely.